tool/tsh/common/hardware_key_test.go

```diff
@@ -0,0 +1,173 @@
+//go:build pivtest
+
+/*
+ * Teleport
+ * Copyright (C) 2023  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package common
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/api/utils/keys"
+	"github.com/gravitational/teleport/lib/auth"
+	"github.com/gravitational/teleport/lib/modules"
+	"github.com/gravitational/teleport/lib/service/servicecfg"
+	testserver "github.com/gravitational/teleport/tool/teleport/testenv"
+)
+
+// TestHardwareKeyLogin tests Hardware Key login and relogin flows.
+func TestHardwareKeyLogin(t *testing.T) {
+	ctx := context.Background()
+
+	testModules := &modules.TestModules{TestBuildType: modules.BuildEnterprise}
+	modules.SetTestModules(t, testModules)
+
+	connector := mockConnector(t)
+
+	alice, err := types.NewUser("alice@example.com")
+	require.NoError(t, err)
+	aliceRole, err := types.NewRole("alice", types.RoleSpecV6{})
```

Comment:
What should we typically expect from a role like this? Would it ordinarily have a target, logins, etc?

Reply:
It would just have the default implicit rules. Since all we need here is to login and attempt `tsh` comands, that's enough.